Add FORMAT UPSERT ENCODE PROTOBUF and related note (#2384)
risingwavelabs/risingwave-docs: versioned_docs/version-1.9/ingest/formats-and-encode-parameters.md

- `JSON`: Data serialized in JSON format in the message queue, compatible with all `FORMAT` options.
- `AVRO`: Data serialized in AVRO format in the message queue, compatible with `FORMAT PLAIN / UPSERT / DEBEZIUM`.
- `Protobuf`: Data serialized in Protobuf format in the message queue, compatible with `FORMAT PLAIN / UPSERT`.
- `CSV`: Data serialized in CSV format in the message queue, compatible with `FORMAT PLAIN`.
- `Bytes`: Data exists in the message queue in raw bytes format, compatible with `FORMAT PLAIN`.

:::note

> - `FORMAT UPSERT ENCODE PROTOBUF` is supported but NOT RECOMMENDED. Because this may disrupt the order of upserts. Because this may disrupt the order of upserts. For more details, see the [documentation of Confluent](https://docs.confluent.io/platform/7.6/control-center/topics/schema.html#c3-schemas-best-practices-key-value-pairs).

Reviewer: ```suggestion
- We support `FORMAT UPSERT ENCODE PROTOBUF` but DON'T RECOMMEND using it, because this may disrupt the order of upserts. Because this may disrupt the order of upserts. For more details, see the [documentation of Confluent](https://docs.confluent.io/platform/7.6/control-center/topics/schema.html#c3-schemas-best-practices-key-value-pairs).
```
